packages/grid/x-data-grid-pro/src/tests/columns.DataGridPro.test.tsx

```diff
@@ -101,6 +101,28 @@ describe('<DataGridPro /> - Columns', () => {
       expect(getCell(1, 0)).toHaveInlineStyle({ width: '110px' });
     });
 
+    /* XXX: Does not reproduce the original issue */
+    it('should not trigger a click on the header after resizing', () => {
+      const columns = [{ field: 'id', minWidth: 100 }, { field: 'brand', width: 100 }];
+      render(
+        <Test
+          columns={columns}
+          initialState={{
+            sorting: {
+              sortModel: [{ field: 'brand', sort: 'asc' }],
+            }
+          }}
+        />
+      );
+      const separator = document.querySelector(`.${gridClasses['columnSeparator--resizable']}`)!;
+      const header = document.querySelector(`.${gridClasses['columnHeader']}`)!;
+      fireEvent.mouseDown(separator, { clientX: 100 });
+      fireEvent.mouseMove(separator, { clientX: 90, buttons: 1 });
+      fireEvent.mouseUp(header);
+      const sortedCell = document.querySelector(`.${gridClasses['columnHeader--sorted']}`);
+      expect(sortedCell?.getAttribute('data-field')).to.equal('brand');
+    });
+
```

Comment:
I haven't been able to reproduce the original issue with the testing framework, so this never fails. What should I do?

Reply:
Missing click event?
```diff
fireEvent.mouseDown(separator, { clientX: 100 });
fireEvent.mouseMove(separator, { clientX: 90, buttons: 1 });
fireEvent.mouseUp(header);
+fireEvent.click(header);
```

Reply:
I've tried adding the click event there, and it's not being prevented . Not sure why, maybe switching to a `setTimeout(..., 0)` would make a difference. I'll try to fiddle with this some more.